hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,235 @@
+import com.hazelcast.nio.Address;
+import com.hazelcast.spi.properties.HazelcastProperties;
+import com.hazelcast.spi.properties.HazelcastProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Created by ihsan on 15/05/17.
+ */
+public class Statistics {
+    private final boolean enabled;
+    private final HazelcastProperties properties;
+    private final ILogger logger = Logger.getLogger(this.getClass());
+    private final HazelcastClientInstanceImpl client;
+    private final boolean enterprise;
+    public static final String PREFIX = "hazelcast.client.statistics";
+
+    public final static String NEAR_CACHE_CATEGORY_PREFIX = "/nearcache";
+
+    private final MetricsRegistry metricsRegistry = new MetricsRegistryImpl(Logger.getLogger(this.getClass()),
+            ProbeLevel.MANDATORY);
+    Map<String, String> staticStats = new HashMap<String, String>(1);
+
+    /**
+     * Use to enable the client statistics collection.
+     * <p/>
+     * The default is false.
+     */
+    public static final HazelcastProperty ENABLED = new HazelcastProperty(PREFIX + ".enabled", false);
+
+    /**
+     * The period in seconds the statictics runs.
+     * <p>
+     * The StatisticsPlugin periodically sending the client statistics to the owner member
+     * <p>
+     * <p>
+     * If set to 0, the plugin is disabled.
+     */
+    public static final HazelcastProperty PERIOD_SECONDS = new HazelcastProperty(PREFIX + ".period.seconds", 60, SECONDS);
+
+    public Statistics(HazelcastClientInstanceImpl client) {
+        this.properties = client.getProperties();
+        this.enabled = properties.getBoolean(ENABLED);
+        this.client = client;
+        this.enterprise = BuildInfoProvider.getBuildInfo().isEnterprise();
+    }
+
+    /**
+     * Registers all client statistics and schedules peridic collection of stats
+     */
+    public void start() {
+        if (!enabled) {
+            return;
+        }
+
+        long periodSeconds = properties.getSeconds(PERIOD_SECONDS);
+        if (periodSeconds < 0) {
+            long defaultValue = Long.parseLong(PERIOD_SECONDS.getDefaultValue());
+            logger.warning(
+                    "Provided client statistics " + PERIOD_SECONDS.getName() + " can not be a negative number. You provided "
+                            + periodSeconds + ". Client will use the default value of " + defaultValue + " instead.");
+            periodSeconds = defaultValue;
+        }
+
+        staticStats.put("/enterprise", Boolean.toString(enterprise));
+        // Register for one time statistics. These are resent only when reconnected to the cluster.
+        registerStaticStatistics();
+
+        registerPeriodicMetrics();
+
+        logger.finest("Client statistics is enabled with period " + periodSeconds + " seconds.");
+
+        // Start task for sending periodic statistics
```

Comment:
If you need a comment for the line below, perhaps it makes more sense to just rename the method to the comment :)

startTaskForSendingPeriodicStatistics.